File: 2-algorithm/intermediateGo/sumOddFiboNums.go
package main

import (
	"fmt"
)

func sumFibs(limit int) int {
	var sum = 2
	var fib = []int{1, 1}
	for i := 2; fib[i-1] < limit; i++ {
		fib = append(fib, fib[i-1]+fib[i-2])
		if fib[i] > limit { // end, limit exceeded
			return sum
		} else if fib[i]%2 != 0 {
			sum += fib[i]
		}
	}
	//fmt.Println(fib)
	return sum
}

func sumOddFiboNums() {
	fmt.Println(sumFibs(1))
	fmt.Println(sumFibs(1000))
	fmt.Println(sumFibs(4000000))
	fmt.Println(sumFibs(4))
	fmt.Println(sumFibs(75024))
	fmt.Println(sumFibs(75025))

}

File: old-v1/1-frontEnd/algorithmMid/sumOddFiboNums.js
/*jslint node: true */
'use strict';

function sumFibs (limit) {
  var total = 1;
  var serie = [0, 1];
  var i = 2;
  while (i <= limit) {
    var sum = serie[i - 1] + serie[i - 2];
    if (sum <= limit) {
      serie.push(sum);
      if (sum % 2 !== 0) {
        total += sum;
      }
    }
    i++;
  }
  return total;
}

console.log(sumFibs(5));
console.log(sumFibs(1000));
console.log(sumFibs(4000000));
console.log(sumFibs(4));
console.log(sumFibs(75024));
console.log(sumFibs(75025));
